GitUI/UserControls/AvatarControl.cs

```diff
@@ -58,6 +58,21 @@ void ClearCache()
             }
         }
 
+        /// <summary>
+        /// Clean up any resources being used.
+        /// </summary>
+        /// <param name="disposing">true if managed resources should be disposed; otherwise, false.</param>
+        protected override void Dispose(bool disposing)
+        {
+            if (disposing && (components != null))
+            {
+                _cancellationTokenSequence.Dispose();
+                components.Dispose();
+            }
```

Comment:
Why dispose the token only if there are components?

Reply:
_cancellationTokenSequence.Dispose(); 
should probably always be called

I was confused when looking at #7365 , but it is likely not the cancellation token there

I have seen some crashes in the debugger when disposing too (no details).

Reply:
I think that is a bug, good find 👍